variadics/src/lib.rs

```diff
@@ -125,6 +127,18 @@ pub trait VariadicExt: Variadic {
     type Reverse: VariadicExt;
     /// Reverses this variadic value.
     fn reverse(self) -> Self::Reverse;
+    /// Reverses an AsRefVar variadic value
+    fn reverse_ref(this: Self::AsRefVar<'_>) -> <Self::Reverse as VariadicExt>::AsRefVar<'_>;
+
+    /// The length of this variadic type
+    fn len(&self) -> usize {
+        Self::LEN
+    }
+
+    /// Checks if this variadic type is empty.
+    fn is_empty(&self) -> bool {
+        Self::LEN == 0
+    }
```

Comment:
When are these needed? Instead of just `Self::LEN`

Reply:
`len` just matches convention. and once you define `len`, rust warns you to define `is_empty`. Which suggests that using `len` convention was good hygiene?